### PR TITLE
e2e-test: add retries to app serve

### PR DESCRIPTION
**Pull request by Rugvip**

### CODE REVIEW

File: packages/e2e-test/src/e2e-test.ts [hunk 1]
```diff
@@ -273,28 +273,36 @@ async function testAppServe(pluginName: string, appDir: string) {
...
+      if (attempts >= 5) {
+        throw new Error(`App serve test failed, ${error}`);
+      }
+      console.log(`App serve failed, trying again, ${error}`);
+      await new Promise(resolve => setTimeout(resolve, 1000));
+    }
   }
 
+  // Kill entire process group, otherwise we'll end up with hanging serve processes
+  killTree(startApp.pid);
```
Comment (freben) [inline packages/e2e-test/src/e2e-test.ts:304]: so this doesn't happen if the max attempts is reached?

File: packages/e2e-test/src/e2e-test.ts [hunk 2]
```diff
@@ -273,28 +273,36 @@ async function testAppServe(pluginName: string, appDir: string) {
...
+    Browser.localhost('localhost', 3000);
 
-    await waitForPageWithText(browser, '/', 'Backstage Service Catalog');
-    await waitForPageWithText(
-      browser,
-      `/${pluginName}`,
-      `Welcome to ${pluginName}!`,
-    );
+    try {
+      const browser = new Browser();
```
Comment (freben) [inline packages/e2e-test/src/e2e-test.ts:282]: does this pick a dynamic port and whatnot?

**Review by freben — APPROVED** [review]

File: packages/e2e-test/src/e2e-test.ts (hunk 2 above)
Comment (Rugvip) [reply packages/e2e-test/src/e2e-test.ts:282]: Nah, this just browses localhost:3000

File: packages/e2e-test/src/e2e-test.ts (hunk 1 above)
Comment (Rugvip) [reply packages/e2e-test/src/e2e-test.ts:304]: Yeah prolly best if it does

**Pull request merged**